### PR TITLE
A0-1540: Verification error

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
This PR introduces error enum for the verification process in pallet `snarcos`. It covers possible errors from `verify(..)` methods in `Groth16`, `GM17` and `Marlin` implementations from `arkworks`, i.e. 
 - `SynthesisError` (for the former two) - here we have only two variants: `MalformedVerifyingKey` and `UnexpectedError`. The first one is just extraction of its counterpart in `SynthesisError`. The second one covers all other variants - they are dedicated for constraint synthesis or proof generation phases.
 - `ark_marlin::Error<ark_poly_commit::Error>` (for the latter one) - here we unpack the `ark_marlin::Error` enum to either `AHPError` or `PolynomialCommitmentError`. In case `IndexTooLarge` happened, we fall back to `UnexpectedError` (this should never happen during verification process)

All the new variants are flat, i.e. they do not contain inner errors. This is because arkworks' errors don't implement basic traits and thus cannot be used in the runtime.

### CODE REVIEW

File: pallets/snarcos/src/systems.rs [hunk 1]
```diff
@@ -1,12 +1,67 @@
...
+            }
+        }
+    }
+}
+
+impl From<ark_marlin::Error<ark_poly_commit::Error>> for VerificationError {
+    fn from(err: ark_marlin::Error<ark_poly_commit::Error>) -> Self {
+        match err {
+            ark_marlin::Error::AHPError(err) => {
+                info!("Encountered AHP error: {err:?}");
```
Comment (maciejzelaszczyk) [inline pallets/snarcos/src/systems.rs:50]: Is there a reason why this is logged as `info` rather than an error? Same for the next case.

**Review by maciejzelaszczyk — APPROVED** [review]
This is very clean indeed! Nifty use of `Into` for error propagation.

File: pallets/snarcos/src/systems.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply pallets/snarcos/src/systems.rs:50]: yes - in that case, we will still push the 'error status' back to the caller, which can be then handled. so in general, the logs shouldn't indicate that there's a program-level error

we are using info channel just to not miss the inner message (`AHPError`) - we cannot pass it to the caller due to unsatisfied traits, so it is the last place when the user can see it

Comment (maciejzelaszczyk) [reply pallets/snarcos/src/systems.rs:50]: OK, gotcha

**Pull request merged**